lib/IDE/CodeCompletion.cpp

```diff
@@ -1630,11 +1630,13 @@ class CodeCompletionCallbacksImpl : public CodeCompletionCallbacks {
 
     // It doesn't type check as a type, so see if it's a qualifying module name.
     if (auto *ITR = dyn_cast<IdentTypeRepr>(ParsedTypeLoc.getTypeRepr())) {
-      SmallVector<ImportDecl::AccessPathElement, 4> AccessPath;
+      // FIXME: Why are we allowing multiple module names? We don't support
+      // submodule qualification.
+      ImportPath::Module::Builder builder;
       for (auto Component : ITR->getComponentRange())
-        AccessPath.push_back({ Component->getNameRef().getBaseIdentifier(),
-                               Component->getLoc() });
-      if (auto Module = Context.getLoadedModule(AccessPath))
+        builder.push_back(Component->getNameRef().getBaseIdentifier(),
+                          Component->getLoc());
+      if (auto Module = Context.getLoadedModule(builder.get()))
```

Comment:
I found the original code here confusing, so it's possible that I've misunderstood it somehow.

Reply:
@brentdax I think you're right. It should probably just early exit when there's more than one component here.

Reply:
I've restructured this to skip this logic unless there is exactly one component in the IdentTypeRepr.